### PR TITLE
All dogs can now eat snacks and dance, including puppy Ian.

### LINKED ISSUE #44230: Pupper Ian does not eat food
Maximum priority this shit, Ian is a hungry boi

### PR DESCRIPTION
**Pull request by Firecage**
## About The Pull Request

~~This PR makes it so that the puppy form of Ian is now its own thing, /corgi/puppy/ian, instead of just a corgi puppy named Ian. This also adds a var, var/ian, to corgis to designate which corgi type is Ian, and moves some other vars to base corgis as well.~~

~~The point of this PR though, is to make it so that the puppy form of Ian is also capable of both eating and dancing as it should be. Do note though, I would appreciate it if someone else ran this to make sure the aging process of Ian is still working correctly.(Well, from my end it seems to be. But due to my PC being a toaster, repeatedly rebooting the server and waiting for the server to setup is a big pain.)~~

This PR now makes it so that all dogs, whether they are corgis or pugs, are now capable of eating snacks and dancing around. The updated version also removed Ian puppy as its own separate mob as my PR originally had since I was unsure if it would cause any issues with the aging system since I couldn't properly test if my PR broke anything or not due to potato PC. HoPs should still be satisfied due to their corgi puppy Ian being able to eat snacks and otherwise enjoy themselves now.

Fixes #44230

## Why It's Good For The Game

It honestly kind of sucked that Ian could not eat nor dance in his puppy form, and thus felt like it wasn't really Ian, but merely a cheap impostor. This aims to fix that.

To add to the above, with the PR now updated, it means that the snack eating is less snowflakey, since it was locked to one specific mob. Now all dogs can eat snacks, and possibly provide endless agony for both chefs and botanists.

## Changelog
:cl:
add: All dogs, including pugs, are now capable of eating snacks and dancing around.
fix: Puppy Ian is now capable of eating and dancing.
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
snacks are bad for puppy ian

**Comment by Firecage** [comment]
No no, only chocolate snacks are. Also maybe snacks tainted by Nar-Sie.

**Comment by cacogen** [comment]
i've always been surprised no one's fleshed out ian more given his popularity with the players. hunger, a dog bowl, dog food, chew toys, fetch, restlessness, etc.. i play hop a lot and there is a lot of downtime.

**Comment by ninjanomnom** [comment]
not a fan of this snowflake for ian honestly, can you just make this regular behavior for corgis?

**Comment by Firecage** [comment]
@ninjanomnom Well, I would actually like it if all pets went after snacks, some having preferences for certain foods.

**Comment by Firecage** [comment]
Will later make a commit for perhaps having all /pet type simple_animals being capable of eating snacks, which I would have already done if I didn't accidentally forget to switch branches and somehow do the work on my master Branch and commit it whereafter I had to fix my master branch.

Still can't figure out why, for example Ian, doesn't give his message when eating food. ("Ian nibbles the tomato.") Doesn't appear to be anything related to this PR, and no idea how to fix it.

Well, will probably remove puppy Ian as a seperate entity since I am unsure if it might cause issues since I can't really frequently normal reboot the local server to test it out due to potato PC, and then make snack eating less restricted to one mob.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -423,7 +423,7 @@
 	fdel(json_file)
 	WRITE_FILE(json_file, json_encode(file_data))
 
-/mob/living/simple_animal/pet/dog/corgi/Ian/Life()
+/mob/living/simple_animal/pet/dog/Life()
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Put this with the other dog procs

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 1 above)
Comment (AnturK) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: @Firecage just move this proc please.

**Review by ninjanomnom — APPROVED** [review]

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 1 above)
Comment (Firecage) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: @AnturK Heh, I would have done so earlier. Rounds on Sybil 2 kiiiinda lasts a long time though.

**Pull request merged**